Sources/Journify/Plugins/other_plugins/IDFACollection.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  IDFACollection.swift
+//  Journify
+//
+//  Created by Mohammed on 12/26/24.
+//
+
+import Foundation
+import UIKit
+import AdSupport
+import AppTrackingTransparency
+
+/**
+ Plugin to collect IDFA values.  Users will be prompted if authorization status is undetermined.
+ Upon completion of user entry a track event is issued showing the choice user made.
+ 
+ Don't forget to add "NSUserTrackingUsageDescription" with a description to your Info.plist.
+ */
+@available(iOS 14, *)
+class IDFACollection: Plugin {
+    let type = PluginType.enrichment
+    weak var analytics: Journify? = nil
+    @Atomic private var alreadyAsked = false
+    
+    func execute<T: RawEvent>(event: T?) -> T? {
+        let status = ATTrackingManager.trackingAuthorizationStatus
+
+        let trackingStatus = statusToString(status)
+        var idfa = fallbackValue
+        var adTrackingEnabled = false
+        
+        if status == .authorized {
+            adTrackingEnabled = true
+            idfa = ASIdentifierManager.shared().advertisingIdentifier.uuidString
+        }
+                
+        var workingEvent = event
+        if var context = event?.context?.dictionaryValue {
+            context[keyPath: "device.adTrackingEnabled"] = adTrackingEnabled
+            context[keyPath: "device.advertisingId"] = idfa
+            context[keyPath: "device.trackingStatus"] = trackingStatus
+            
+            workingEvent?.context = try? JSON(context)
+        }
+
+        return workingEvent
+    }
+}
+
+@available(iOS 14, *)
+extension IDFACollection: iOSLifecycle {
+    func applicationDidBecomeActive(application: UIApplication?) {
+        let status = ATTrackingManager.trackingAuthorizationStatus
+
+        if status == .notDetermined && !alreadyAsked {
+            // we don't know, so should ask the user.
+            alreadyAsked = true
+            DispatchQueue.main.async { [weak self] in
+                guard let self else { return }
+                askForPermission()
```

Comment:
question: Do we automatically ask for permission here? in other SDK we have a flag to ask permission or just skip if the flag is false.

Reply:
@amineck To use the IDFA, the client should make sure to include "NSUserTrackingUsageDescription" with a description in their Info.plist. In the askForPermission function, I check whether this property has already been added. If it’s there, I display the popup; if not, I don’t.